### PR TITLE
LPS-176161 | master

### PR DESCRIPTION
**Pull request by melodywu**
https://issues.liferay.com/browse/LPS-176161

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/macros/CookiePreferenceHandlingAdmin.macro [hunk 1]
```diff
@@ -1,5 +1,35 @@
 definition {
 
+	@description = "Disable Cookie Preference handling"
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/macros/CookiePreferenceHandlingAdmin.macro:3]: Please expand the description indicating it disables the cookie settings at Instance level

File: portal-web/test/functional/com/liferay/portalweb/macros/CookiePreferenceHandlingAdmin.macro [hunk 2]
```diff
@@ -54,6 +84,31 @@ definition {
 		CookiePreferenceHandlingAdmin.saveConfiguration();
 	}
 
+	@description = "Enable Cookie Preference handling with Explicit cookie setting disabled"
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/macros/CookiePreferenceHandlingAdmin.macro:87]: Extend description to indicate instance level

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+				value1 = "SITE_LEVEL_1 SITE_LEVEL_3");
+
+			CookieConsentPanel.verifyCookiesForCookieCategoriesAreNotPresent();
+
+			CookieHelper.checkSelectedCookieIsNotPresent(
+				cookieName = "GUEST_LANGUAGE_ID",
+				expectedCookieValue = "en_US");
+		}
+
+		task ("And: End User first visits the instance,Verify cookie banner with instance level is present.") {
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase:241]: space after ,

**Review by gaborlovasdotcom — COMMENTED** [review]
Looks good, only small formatting changes required

File: portal-web/test/functional/com/liferay/portalweb/macros/CookiePreferenceHandlingAdmin.macro (hunk 1 above)
Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/macros/CookiePreferenceHandlingAdmin.macro:3]: Done

File: portal-web/test/functional/com/liferay/portalweb/macros/CookiePreferenceHandlingAdmin.macro (hunk 2 above)
Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/macros/CookiePreferenceHandlingAdmin.macro:87]: done

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesUseCase.testcase [hunk 4]
```diff
@@ -80,6 +82,127 @@ definition {
...
+				expectedCookieValue = "en_US");
+		}
+	}
+
+	@description = "This is a use case for LPS-170904 TC-8. Verify only functional cookies are set in Site level enable then disable, when the user isn't a first-time visitor."
+	@priority = 5
+	test VerifyFunctionalCookiesForNotFirstVisitInInstanceLevelEnableThenDisable {
+		property test.name.skip.portal.instance = "CookiesSiteLevelUseCase#VerifyFunctionalCookiesForNotFirstVisitInInstanceLevelEnableThenDisable";
+
+		task ("Given: Site Administrator enabled agreed on the cookie usage,then disable the cookie preference handling in site level.") {
```
[the remark this answers is not in the document]
Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesUseCase.testcase:162]: done

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase (hunk 3 above)
Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase]: Done

**Comment by melodywu** [comment]
ci:test:sf

> Looks good, only small formatting changes are required
> 
> Please also check ci:test:cookies:nocompile for test execution to make sure the tests are present in this suite as well.

11 testcases are in this suite

**Comment by JoyceWang08211** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesUseCase.testcase [hunk 5]
```diff
@@ -2,6 +2,7 @@
 definition {
 
 	property ci.retries.disabled = "true";
+	property custom.properties = "feature.flag.LPS-142518=true${line.separator}jsonws.web.service.paths.excludes=";
```
Comment (ling-alan-huang) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesUseCase.testcase:5]: @melodywu  
If the changes contains `feature.flag`, SF will enforce checking `portal-impl/src/portal.properties`.
I send the fix here: https://github.com/ling-alan-huang/liferay-portal/pull/1669

Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesUseCase.testcase:5]: thank you Alan.

**Comment by melodywu** [comment]
create a new pull [#1025 ](https://github.com/liferay-appsec/liferay-portal/pull/1025)

**Pull request closed without merging**